modules/javafx.controls/src/main/java/javafx/scene/control/Skinnable.java

```diff
@@ -36,23 +36,30 @@
  */
 public interface Skinnable {
     /**
-     * Skin is responsible for rendering this {@code Control}. From the
-     * perspective of the {@code Control}, the {@code Skin} is a black box.
-     * It listens and responds to changes in state in a {@code Control}.
+     * Skin is responsible for rendering this {@code Skinnable}. From the
+     * perspective of the {@code Skinnable}, the {@code Skin} is a black box.
+     * It listens and responds to changes in state in a {@code Skinnable}.
      * <p>
-     * There is a one-to-one relationship between a {@code Control} and its
+     * There is typically a one-to-one relationship between a {@code Skinnable} and its
```

Comment:
When isn't it a 1-to-1 relationship? Is it just for the special case of `PopupControl`?

Reply:
Yes. All Skins derived by SkinBase must have a 1:1 relationship, while `getSkinnable()` for `PopupControl` is not really used.

From my other comment:
It is a bit weird as Popups are not using the `getSkinnable` method it looks like. And also the `getSkinnable` method javadoc states: Gets the Skinnable to which this Skin is assigned. which is not true for the ComboBox Popup (as it sets the ComboBox as the skinnable) - while it makes sense that the ComboBox is set as styleable parent.

And:
So it is even more weird:
`getSkinnable()` is always called in Control skins, `getNode()` is only called in `Control.getSkinNode()`, PaginationSkin and TableColumnHeader (so not very much).
`getNode()` is always called for Popups (no `getSkinnable()` is called).

The only exception is the internal class `InputFieldSkin`, where the node is an 'inner TextField' and the skinnable the control where it is installed on.